src/testing.rs

```diff
@@ -0,0 +1,148 @@
+//! Enable the test_helpers feature to expose helper methods to build
+//! mock response structures for testing your code that uses the imap crate
+//!
+//! To use add a dev-dependency on the imap extension adding the feature "test_helpers"
+//! e.g.
+//!
+//! [features]
+//! tls-rustls = ["imap/rustls-tls"]
+//! default = ["tls-rustls"]
+//!
+//! [dependencies]
+//! imap = { version = "3.0", default-features = false }
+//!
+//! [dev-dependencies]
+//! # mirror the same configuration your dependencies and add test_helpers
+//! imap = { version = "3.0", default-features = false, features = ["test_helpers"] }
+//!
+use crate::error::Result;
+use crate::extensions::list_status::ExtendedNames;
+use crate::types::{
+    AclResponse, Capabilities, Fetches, ListRightsResponse, MyRightsResponse, Names, QuotaResponse,
+    QuotaRootResponse,
+};
+use std::sync::mpsc;
+
+/// Builds an [`Capabilities`] based on the provided input
+///
+/// Example input.
+///
+/// let input = b"\
+/// * CAPABILITY IMAP4rev1 STARTTLS AUTH=GSSAPI LOGINDISABLED\r\n\
+/// ";
+pub fn build_capabilities(input: Vec<u8>) -> Result<Capabilities> {
```

Comment:
I think it'd be nice if these were in submodules instead so that we have a place to add builders later. I'm thinking:
```rust
mod capabilities {
  pub fn parse(input: Vec<u8>) -> Result<Capabilities>;
}
```
I also think we should use `impl Into<Vec<u8>>` instead so that byte strings will just work and not require `.into()`.

And finally, since this is just for testing (and to help ensure that's the case), what do you think of always `.unwrap`ing inside of these rather than exposing the result. Makes it less attractive to use them for anything else.

Reply:
sounds good on all points.   I'll make the adjustments.